### ISSUE TITLE
[Feature] Add AmneziaWG Protocol

### ISSUE THREAD

**Issue by leaked** [MetaCubeX/mihomo#1507]
### Verify steps

- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的请求 I have searched on the [issue tracker](……/) for a related feature request. 
- [X]  我已经仔细看过 [Documentation](https://wiki.metacubex.one/) 并无法找到这个功能 I have read the [documentation](https://wiki.metacubex.one/) and was unable to solve the issue. 

### Description

Hello
Please add this protocol to the program
https://github.com/amnezia-vpn/amneziawg-go

https://github.com/amnezia-vpn/amneziawg-windows-client
https://github.com/amnezia-vpn/amneziawg-android

Thanks

**Comment by wwqgtxx** [comment]
done https://github.com/MetaCubeX/mihomo/commit/781b783346f58bbba3370ed4708a806e4b71c24f

**Comment by leaked** [comment]
> done [781b783](https://github.com/MetaCubeX/mihomo/commit/781b783346f58bbba3370ed4708a806e4b71c24f)

I built the latest commit, but I couldn't connect to the server. Did you test it yourself?

**Comment by wwqgtxx** [comment]
Did you write a wrong config file?

 `amnezia-wg-option` is under the top level config instead of inside `peers`.

**Comment by leaked** [comment]
> Did you write a wrong config file?
> 
> `amnezia-wg-option` is under the top level config instead of inside `peers`.

I checked several times and everything seems to be correct, but the problem is from somewhere else

```
  - name: "test"
    type: wireguard
    server: 1.1.1.1
    port: 443
    private-key: eCtXsJZ27+4PbhDkHnB923tkUn2Gj59wZw5wFA75MnU=
    public-key: Cr8hWlKvtDt7nrvf+f0brNQQzabAqrjfBvas9pmowjo=
    pre-shared-key: 31aIhAPwktDGpH4JDhA8GNvjFXEf/a6+UaQRyOAiyfM=
    ip: 172.16.0.2
    allowed-ips: ["0.0.0.0/0", "::/0"]
    remote-dns-resolve: true
    dns: [1.1.1.1, 1.0.0.1]
    persistent-keepalive: 25
    amnezia-wg-option:
      jc: 5
      jmin: 500
      jmax: 501
      s1: 30
      s2: 40
      h1: 123456
      h2: 67543
      h3: 123123
      h4: 32345
    mtu: 1420
    udp: true
```
Can we connect on a platform like Telegram so I can send you the server information for testing?

**Comment by wwqgtxx** [comment]
Maybe you can send the real configuration to the gmail of the same name, and attach the configuration file that can be linked normally in the amneziawg official client.

**Comment by leaked** [comment]
> Maybe you can send the real configuration to the gmail of the same name, and attach the configuration file that can be linked normally in the amneziawg official client.

I sent it to you, thank you.

**Comment by wwqgtxx** [comment]
fixed in: https://github.com/MetaCubeX/mihomo/commit/6d48d17d43a4ab303c50a1f9cb085b2e690381a0

**Comment by X-49** [comment]
> fixed in: [6c03830](https://github.com/MetaCubeX/mihomo/commit/6c0383026eeeab77853d048fc1434b3529e468a8)

Thank you so much for this, went back to using Mihomo with FlClash instead of Hiddify & AmneziaVPN. Your core is the most advanced in terms of functionality, the WireGuard obfuscation capabilities are very valuable.

By the way, Hiddify had a number of additional options for [WireGuard obfuscation](https://hiddify.com/app/How-to-use-WARP-on-Hiddify-App/), are they any different from AmneziaWG? Maybe they should be implemented in the kernel too, that would be great.
